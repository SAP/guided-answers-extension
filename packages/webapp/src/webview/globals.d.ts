interface Window {
    vscode: {
        postMessage: (message: string | JSON) => void;
    };
}
