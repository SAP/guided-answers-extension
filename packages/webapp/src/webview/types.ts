import type {
    GuidedAnswerNode,
    GuidedAnswerTree,
    GuidedAnswerTreeSearchResult
} from '@sap/guided-answers-extension-types';

export interface AppState {
    loading: boolean;
    query: string;
    guidedAnswerTreeSearchResult: GuidedAnswerTreeSearchResult;
    activeGuidedAnswerNode: GuidedAnswerNode[];
    activeGuidedAnswer?: GuidedAnswerTree;
    betaFeatures: boolean;
    searchResultCount: number;
    guideFeedback: null | boolean;
    selectedProductFilters: string[];
    selectedComponentFilters: string[];
    pageSize: number;
    feedbackStatus: boolean;
    feedbackResponse: boolean;
}
