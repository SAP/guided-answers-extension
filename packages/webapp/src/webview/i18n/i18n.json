{
    "GUIDED_ANSWER_HEADER": "Guide",
    "SEARCH_GUIDED_ANSWERS": "Search Guided Answers",
    "ALL_ANSWERS": "Home",
    "STEP_BACK": "Step back",
    "RESTART": "Restart",
    "GUIDED_ANSWERS": "Guided Answers",
    "NO_ANSWERS_FOUND": "No answers found",
    "PLEASE_MODIFY_SEARCH": "Please modfiy search",
    "THIS_SOLVED_MY_ISSUE": "This solved my issue",
    "THIS_DID_NOT_SOLVED_MY_ISSUE": "This did not solve my issue",
    "THANK_YOU_TEXT": "We are glad to hear that your issue has been resolved and we hope you enjoy using the Guided Answers application.",
    "THANKS": "Thanks!",
    "HOME": "Home",
    "SEND": "Send",
    "CLOSE": "Close",
    "ISSUE_IS_NOT_RESOLVED": "Issue is not resolved",
    "WE_ARE_SORRY_TO_HEAR_THAT_YOUR_ISSUE_IS_NOT_YET_RESOLVED": "We are sorry to hear that your issue is not yet resolved.",
    "PLEASE_TELL_US_IF_THIS_ANSWER_WAS_HELPFUL": "Please tell us if this answer was helpful",
    "THERE_ARE_SEVERAL_OPTIONS_FOR_GETTING_FURTHER_ASSISTANCE": "There are several options for getting further assistance:",
    "SEARCH_FOR_ANOTHER_GUIDED_ANSWER": "Search for another Guided Answer",
    "START_AN_EXPERT_CHAT": "Start an Expert Chat",
    "SCHUEDULE_AN_EXPERT": "Schedule an Expert",
    "OPEN_AN_INCIDENT": "Open an Incident",
    "ASK_THE_SAP_COMMUNITY": "Ask the SAP Community",
    "FEEDBACK": "Feedback",
    "FEEDBACK_DIALOG_TITLE": "Is this content helpful?",
    "FEEDBACK_DIALOG_SUBTEXT": "If you have suggestions on how to improve this content we would love to hear them!",
    "FEEDBACK_DIALOG_DISCLAIMER": "Your feedback is anonymous, we do not collect any personal data.",
    "FEEDBACK_DIALOG_SUGGESTION": "Your suggestion",
    "FEEDBACK_SENT_DIALOG_TITLE": "Message sent",
    "FEEDBACK_SENT_DIALOG_SUBTEXT": "Thanks for the feedback!"
}
