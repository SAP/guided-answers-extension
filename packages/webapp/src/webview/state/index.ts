export { actions, store } from './store';
export { getInitialState } from './reducers';
