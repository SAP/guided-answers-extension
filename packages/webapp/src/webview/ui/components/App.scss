html {
    height: 100%;
}
* {
    font-family: var(--vscode-font-family);
    -webkit-font-smoothing: antialiased;
}

h1 {
    font-weight: 300;
    color: var(--vscode-foreground);
    font-size: 24px;
}

h2 {
    font-size: 13px;
}

.guided-answer {
    &__header {
        overflow: hidden;
        position: fixed; /* Set the navbar to fixed position */
        top: 0; /* Position the navbar at the top of the page */
        width: 100%;
        z-index: 10;
        background-color: var(--vscode-editor-background);
        &__title {
            font-weight: 600;
            margin: 0px;
            color: var(--vscode-settings-headerForeground);
            line-height: 29px;
        }
        &__subtitle {
            color: var(--vscode-editorCodeLens-foreground, #999);
            font-size: 2em;
        }
        &__goback {
            height: 26px;
            margin-top: 17px;
        }
        &__logoAndTitle {
            display: flex;
            align-items: center;
            margin-top: 40px;
        }
        &__searchField {
            display: flex;
            align-items: center;
            flex-direction: column;
            margin-top: -28px;
            margin-bottom: 40px;
        }
    }
    &__container {
        margin-top: 111px;
        padding-bottom: 10px;
    }
    &__tree {
        width: 100%;
        height: auto;
        cursor: pointer;
        text-align: left;
        display: flex;
        border: none;
        &:hover {
            background-color: var(--vscode-list-hoverBackground);
        }

        &__ul {
            padding: 15px;
            &h3 {
                margin-bottom: 10px;
            }
        }
        &__title {
            font-size: 14px;
            color: var(--vscode-settings-headerForeground);
            margin-bottom: 10px;
            margin-top: 0px;
        }
        &__desc {
            color: var(--vscode-foreground);
            height: auto;
            font-size: 13px;
        }
    }
}

.tree-search-field {
    width: 33%;
    min-width: 250px;
}

.tree-search-icon {
    margin-top: 5px;
    text-align: center;
}

.tree-item {
    margin: 0;
}

#sap-logo {
    width: 59px;
}

.striped-list {
    padding-left: 0px;
    margin-top: 0px;
}

ul.striped-list > li:nth-of-type(odd) {
    background-color: var(--vscode-breadcrumbPicker-background);
    padding-left: 0px;
}

li {
    background-color: var(--vscode-editor-background);
    list-style-type: none;
}
