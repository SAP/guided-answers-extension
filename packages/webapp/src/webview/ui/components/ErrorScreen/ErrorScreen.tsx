import React from 'react';
import './ErrorScreen.scss';
import ErrorIcon from './no-answers-found.svg';

/**
 * @param props - props for error screen component
 * @param props.title - the screen title
 * @param props.subtitle - the screen subtitle
 *@returns - react element to show an error page when there are no answers found
 */
export function ErrorScreen(props: Readonly<{ title: string; subtitle: string }>) {
    return (
        <div className="error-screen">
            <div className="error-screen__objects">
                <h1 className="error-screen__object__title">{props.title}</h1>
                <h2 className="error-screen__object__subtitle">{props.subtitle}</h2>
                <span className="error-screen__object__logo">
                    <ErrorIcon />
                </span>
            </div>
        </div>
    );
}
