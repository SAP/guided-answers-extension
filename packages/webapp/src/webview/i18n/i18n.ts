import { use } from 'i18next';
import { initReactI18next } from 'react-i18next';
import * as i18nEn from './i18n.json';

export { i18n } from 'i18next';

export function initI18n(): void {
    use(initReactI18next)
        .init({
            resources: {
                en: {
                    translation: i18nEn
                }
            },
            lng: 'en',
            fallbackLng: 'en',
            interpolation: {
                escapeValue: false
            }
        })
        .catch((error) => console.error(error));
}
