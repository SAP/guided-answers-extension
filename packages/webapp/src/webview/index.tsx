import React from 'react';
import { render } from 'react-dom';
import { Provider } from 'react-redux';
import { App } from './ui/components/App';
import { actions, store } from './state';
import { initI18n } from './i18n';

/**
 * Initialization of i18n, icons
 */
initI18n();

/**
 * Render app
 */
render(
    <Provider store={store}>
        <App />
    </Provider>,
    document.getElementById('root') as HTMLElement
);

/**
 * Fire action which is captured in hosting app (ide-extension) to signal that we are ready to receive messages
 */
actions.webviewReady();

/**Once webview is ready we focus to search field */
document.getElementById('search-field')?.focus();
