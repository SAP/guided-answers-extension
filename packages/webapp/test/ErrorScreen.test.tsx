import React from 'react';
import { ErrorScreen } from '../src/webview/ui/components/ErrorScreen';
import { initI18n } from '../src/webview/i18n';
import { render } from '@testing-library/react';

describe('<ErrorScreen />', () => {
    initI18n();

    it('Should render a ErrorScreen component', () => {
        const { container } = render(<ErrorScreen />);
        expect(container).toMatchSnapshot();
    });
});
