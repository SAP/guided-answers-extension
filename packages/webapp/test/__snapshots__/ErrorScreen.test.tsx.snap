// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorScreen /> Should render a ErrorScreen component 1`] = `
<div>
  <div
    class="error-screen"
  >
    <div
      class="error-screen__objects"
    >
      <h1
        class="error-screen__object__title"
      >
        TITLE
      </h1>
      <h2
        class="error-screen__object__subtitle"
      >
        SUBTITLE
      </h2>
      <span
        class="error-screen__object__logo"
      >
        <svg />
      </span>
    </div>
  </div>
</div>
`;
